src/formatDate.js

```diff
@@ -7,8 +7,33 @@
  *
  * @returns {string}
  */
-function formatDate(date, fromFormat, toFormat) {
-  // write code here
+function formatDate(date, fromFormat, toFormat, cutoffYear = 30) {
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateParts = date.split(fromSeparator);
+  const dateSet = {};
+
+  for (let i = 0; i < dateParts.length; i++) {
+    dateSet[fromFormat[i]] = dateParts[i];
+  }
+
+  const newFormat = [];
+
+  for (const key of toFormat) {
+    if (dateSet[key]) {
+      newFormat.push(dateSet[key]);
+    } else if (key === 'YY') {
+      newFormat.push(dateSet.YYYY.slice(2));
+    } else if (key === 'YYYY') {
+      if (dateSet.YY < cutoffYear) {
+        newFormat.push('20' + dateSet.YY);
+      } else {
+        newFormat.push('19' + dateSet.YY);
```

Comment:
The logic assumes that `dateSet.YY` is always available, which might not be the case if the input format does not include 'YY'. Consider adding checks to handle scenarios where 'YY' is not part of the `fromFormat`.